metrics.go

```diff
@@ -50,6 +50,10 @@ func (c *sidekickCollector) Describe(ch chan<- *prometheus.Desc) {
 
 // Collect is called by the Prometheus registry when collecting metrics.
 func (c *sidekickCollector) Collect(ch chan<- prometheus.Metric) {
+	// automatically read the global stats
+	// Read/Write Lock is not required here
+	globalConnStatsRWMutex.RLock()
+	defer globalConnStatsRWMutex.RUnlock()
 	for _, c := range globalConnStats {
```

Comment:
Let's minimize the time we hold the lock. All values from the slice are loaded atomically anyway.
```suggestion
	globalConnStatsRWMutex.Lock()
	conns := globalConnStats
	globalConnStatsRWMutex.Unlock()
	for _, c := range  conns {
```